src/common/transformations/src/transformations/common_optimizations/transpose_sinking_concat.cpp

```diff
@@ -32,18 +32,26 @@ ov::pass::TransposeSinkingConcatForward::TransposeSinkingConcatForward() {
         auto main_node = main_node_output.get_node_shared_ptr();
 
         TransposeInputsInfo transpose_input_info = GetFirstTransposeInput(main_node);
+        auto concat_node = as_type_ptr<Concat>(main_node);
+        auto concat_axis = concat_node->get_concatenation_axis();
+        if (concat_axis < 0) {
+            return false;
+        }
```

Comment:
why do we decline to handle this case? Can we normalize axis for the concatenation to have non-negative?

Reply:
if `concat_node->get_concatenation_axis();` returns negative value, it means that normalization has failed already (see validate_and_infer_type method of Concat op). Most probably rank is dynamic